lib/Doctrine/DBAL/Types/JsonArrayType.php

```diff
@@ -70,12 +51,5 @@ public function getName()
     {
         return Type::JSON_ARRAY;
     }
-
-    /**
-     * {@inheritdoc}
-     */
-    public function requiresSQLCommentHint(AbstractPlatform $platform)
```

Comment:
the issue is that both the json_array and the json type are now mapped to the native JSON type, meaning it can break schema changes for platforms using the native type. You need to distinguish them to avoid endless schema changes for them